### PR TITLE
Remove "generate normal maps" feature

### LINKED ISSUE #10307: "Generate Normal Maps" should not be in the settings menu
##### Summary
When people enable additional graphics features, they expect better graphics, not worse
The original issue was that months / years ago I enabled it, forgot about it, now I open the game again and everything looks like s***
I'm sorry but it looks absolutely atrocious

Disabled:
immagine

Enabled:
immagine

### PR DESCRIPTION
**Pull request by hecktest**
Fixes #10307

## To do

- [x] Erase all traces of normal "generation" from fragment shaders
- [x] Remove the "feature" from the engine and default config
- [x] Remove any leftover documentation of it
- [x] Test

### CODE REVIEW

File: client/shaders/nodes_shader/opengl_fragment.glsl [hunk 1]
```diff
@@ -76,11 +76,6 @@ float intensity(vec3 color)
 	return (color.r + color.g + color.b) / 3.0;
 }
 
-float get_rgb_height(vec2 uv)
-{
-	return intensity(texture2D(baseTexture, uv).rgb);
-}
```
Comment (HybridDog) [inline client/shaders/nodes_shader/opengl_fragment.glsl:82]: please also remove the now unused `intensity` function here and in the object shader code

**Comment by rubenwardy** [comment]
You don't need to worry about touching the translations - it will automatically be removed when they are next updated

**Comment by hecktest** [comment]
But there is one line in the translations that requires modification, not just removal:
`"Enables bumpmapping for textures. Normalmaps need to be supplied by the "
"texture pack\n"
"or need to be auto-generated.\n"`

**Comment by rubenwardy** [comment]
You modify the source of those translations by editing settingtypes.txt. The translation template is generated from *.lua, *.cpp, and settingtypes.txt

**Comment by hecktest** [comment]
So you're saying that the old string will be discarded and translators will just have to rewrite that line?

**Comment by rubenwardy** [comment]
yeah, if the source string is changed then it's as if it were deleted and recreated, afaik

**Comment by hecktest** [comment]
Done. Please tell me if I missed anything or ripped out too much.

**Comment by paramat** [comment]
Just so you know in future: Changes to minetest.conf.example are not necessary, as that file is regularly auto-generated from settingtypes.txt, so changes are wasted work. However, the changes are harmless so it is not essential to remove them from the PR.

**Comment by hecktest** [comment]
Why isn't it gitignored then?

**Comment by paramat** [comment]
Good point, maybe we should.

**Review by rubenwardy — APPROVED** [review]

**Review by sfan5 — APPROVED** [review]

**Comment by Zughy** [comment]
(bump, as this can be merged: 2 approvals)

**Pull request merged**